sourced/ml/algorithms/uast_struct_to_bag.py

```diff
@@ -28,44 +36,49 @@ def _uast2seq(self, root, walk):
                 nodes[id(child)].extend(child.children)
                 stack.append(child)
             else:
-                walk.append(stack.pop())
+                sequence.append(stack.pop())
+        return sequence
 
-    def uast_to_bag(self, uast):
+    def __call__(self, uast):
         bag = defaultdict(int)
-        seq = []
-        self._uast2seq(uast, seq)
+        node_sequence = self._uast2sequence(uast)
 
         # convert to str - requirement from wmhash.BagsExtractor
-        seq = "".join([self.node2ind(n) for n in seq])
+        node_sequence = "".join([self.node2index[n] for n in node_sequence])
 
-        if isinstance(self.seq_len, int):
-            seq_lens = [self.seq_len]
-        else:
-            seq_lens = self.seq_len
-
-        for seq_len in seq_lens:
-            for i in range(0, len(seq) - seq_len + 1, self.stride):
-                bag[seq[i:i + seq_len]] += 1
+        for seq_len in self._seq_lens:
+            for i in range(0, len(node_sequence) - seq_len + 1, self._stride):
+                bag[node_sequence[i:i + seq_len]] += 1
         return bag
 
-    def node2ind(self, node):
```

Comment:
now we have `_node2index` dict-like object. As was earlier coded in `UastIds2Bag`. 
Want to make it consistent.